### PR TITLE
Use shared CTime::GetCTimeInstance() instead of CTime::init()

### PR DESCRIPTION
**Pull request by lukas-phaf**

### CODE REVIEW

File: CCDFDataModel/CCDFPNGIO.cpp [hunk 1]
```diff
@@ -152,9 +152,8 @@ int CDFPNGReader::open(const char *fileName) {
         timeVariable->setAttributeText("units", "seconds since 1970-01-01 0:0:0");
         timeVariable->setAttributeText("standard_name", "time");
         timeVariable->allocateData(1);
-        CTime ctime;
-        ctime.init(timeVariable);
-        ((double *)timeVariable->data)[0] = ctime.dateToOffset(ctime.freeDateStringToDate(pngRaster->headers[j].value));
+        CTime *ctime = CTime::GetCTimeInstance(timeVariable);
+        ((double *)timeVariable->data)[0] = ctime->dateToOffset(ctime->freeDateStringToDate(pngRaster->headers[j].value));
```
Comment (maartenplieger) [inline CCDFDataModel/CCDFPNGIO.cpp:160]: Can you add a check in case ctime == nullptr?

File: CCDFDataModel/CCDFPNGIO.cpp [hunk 2]
```diff
@@ -169,9 +168,8 @@ int CDFPNGReader::open(const char *fileName) {
         referenceTimeVariable->setAttributeText("units", "seconds since 1970-01-01 0:0:0");
         referenceTimeVariable->setAttributeText("standard_name", "forecast_reference_time");
         referenceTimeVariable->allocateData(1);
-        CTime ctime;
-        ctime.init(referenceTimeVariable);
-        ((double *)referenceTimeVariable->data)[0] = ctime.dateToOffset(ctime.freeDateStringToDate(pngRaster->headers[j].value));
+        CTime *ctime = CTime::GetCTimeInstance(referenceTimeVariable);
+        ((double *)referenceTimeVariable->data)[0] = ctime->dateToOffset(ctime->freeDateStringToDate(pngRaster->headers[j].value));
```
Comment (maartenplieger) [inline CCDFDataModel/CCDFPNGIO.cpp:180]: Same check here ctime == nullptr

File: CCDFDataModel/testccdfdatamodel.cpp [hunk 3]
```diff
@@ -67,18 +67,17 @@ int testCTimeInit(CDF::Variable *testVar, const char *testDate) {
 }
 
 int testCTime2Init(CDF::Variable *testVar, double valueToCheck) {
-  CTime adagucTime;
-  adagucTime.init(testVar);
+  CTime *adagucTime = CTime::GetCTimeInstance(testVar);
 
   try {
-    double startGraphTime = adagucTime.dateToOffset(adagucTime.freeDateStringToDate("2014-08-08T11:22:33Z"));
+    double startGraphTime = adagucTime->dateToOffset(adagucTime->freeDateStringToDate("2014-08-08T11:22:33Z"));
```
Comment (maartenplieger) [inline CCDFDataModel/testccdfdatamodel.cpp:73]: Same check here ctime == nullptr

File: adagucserverEC/CConvertADAGUCPoint.cpp [hunk 4]
```diff
@@ -970,7 +971,7 @@ int CConvertADAGUCPoint::convertADAGUCPointData(CDataSource *dataSource, int mod
               }
               // obsTimeData
 
-              lastPoint->paramList.push_back(CKeyValue(key, description, obsTime.dateToISOString(obsTime.getDate(obsTimeData[pPoint])).c_str()));
+              lastPoint->paramList.push_back(CKeyValue(key, description, obsTime->dateToISOString(obsTime->getDate(obsTimeData[pPoint])).c_str()));
```
Comment (maartenplieger) [inline adagucserverEC/CConvertADAGUCPoint.cpp:974]: Same check here ctime == nullptr? Previousely getDate would throw an exception.

File: adagucserverEC/CConvertGeoJSON.cpp [hunk 5]
```diff
@@ -432,12 +432,11 @@ void CConvertGeoJSON::getDimensions(CDFObject *cdfObject, json_value &json, bool
...
               CDF::Variable timeVarHelper;
               timeVarHelper.setAttributeText("units", "seconds since 1970-1-1");
-              CTime timeHelper;
-              timeHelper.init(&timeVarHelper);
+              CTime *timeHelper = CTime::GetCTimeInstance(&timeVarHelper);
 
               double timeOffset;
               if (timeVal.length() > 0) {
-                timeOffset = timeHelper.dateToOffset(timeHelper.freeDateStringToDate(timeVal.c_str()));
+                timeOffset = timeHelper->dateToOffset(timeHelper->freeDateStringToDate(timeVal.c_str()));
```
Comment (maartenplieger) [inline adagucserverEC/CConvertGeoJSON.cpp:442]: Same check here timeHelper == nullptr

File: adagucserverEC/CConvertGeoJSON.cpp [hunk 6]
```diff
@@ -513,12 +512,11 @@ void CConvertGeoJSON::getDimensions(CDFObject *cdfObject, json_value &json, bool
...
```
Comment (maartenplieger) [inline adagucserverEC/CConvertGeoJSON.cpp:522]: Same check here timeHelper == nullptr

File: adagucserverEC/CConvertTROPOMI.cpp [hunk 7]
```diff
@@ -180,11 +180,10 @@ int CConvertTROPOMI::convertTROPOMIHeader(CDFObject *cdfObject, CServerParams *)
   varT->setAttributeText("standard_name", "time");
   try {
     varT->setAttributeText("units", productT->getAttribute("units")->toString().c_str());
-    CTime myTime;
-    myTime.init(productT);
+    CTime *myTime = CTime::GetCTimeInstance(productT);
```
Comment (maartenplieger) [inline adagucserverEC/CConvertTROPOMI.cpp:183]: Same check here myTime == nullptr

**Review by belentorrente — APPROVED** [review]
Looks good to me

**Pull request merged**